t/t0001-basic.t

```diff
@@ -161,24 +161,24 @@ test_expect_success 'flux-start -o,--setattr ATTR=VAL can set broker attributes'
 	test $ATTR_VAL -eq 42
 '
 test_expect_success 'tbon.endpoint can be read' '
-	ATTR_VAL=`flux start flux getattr tbon.endpoint` &&
-        echo $ATTR_VAL | grep "^tcp"
+	ATTR_VAL=`flux start -s2 flux getattr tbon.endpoint` &&
+        echo $ATTR_VAL | grep "://"
 '
 test_expect_success 'tbon.endpoint can be set and %h works' '
-	ATTR_VAL=`flux start -o,--setattr=tbon.endpoint='tcp://%h:*' flux getattr tbon.endpoint` &&
+	ATTR_VAL=`flux start -s2 -o,--setattr=tbon.endpoint='tcp://%h:*' flux getattr tbon.endpoint` &&
         echo $ATTR_VAL | grep "^tcp" &&
         ! echo $ATTR_VAL | grep "%h"
 '
 test_expect_success 'tbon.endpoint with %B works' '
-	ATTR_VAL=`flux start -o,--setattr=tbon.endpoint='ipc://%B/req' flux getattr tbon.endpoint` &&
+	ATTR_VAL=`flux start -s2 -o,--setattr=tbon.endpoint='ipc://%B/req' flux getattr tbon.endpoint` &&
         echo $ATTR_VAL | grep "^ipc" &&
         ! echo $ATTR_VAL | grep "%B"
 '
 test_expect_success 'tbon.endpoint fails on bad endpoint' '
-	! flux start -o,--setattr=tbon.endpoint='foo://bar' flux getattr tbon.endpoint
+	! flux start -s2 -o,--setattr=tbon.endpoint='foo://bar' flux getattr tbon.endpoint
```

Comment:
Since you are editing these tests anyway, mind switching from `!` to `test_must_fail`? See docs for [`test_must_fail`](https://github.com/mlafeldt/sharness/blob/master/API.md#test_must_fail) for a description of why.

Reply:
Will do.  Actually that last one fails under `test_must_fail` because rank 1 broker has to be killed by `flux-start`.  I'll add a comment for housekeepers of the future.